### PR TITLE
Switch analytics routing through Firebase

### PR DESCRIPTION
**Pull request by ilikesymmetry**
# Overview
- changes pod dependency to use `FirebaseAnalytics` instead of deprecated `Crashlytics` and `Fabric`
- merges log payload file and analytics class files
- renames "Payload" as "Event" for clearer context of what we are sending

# Next steps
- Rename repository `analytics-ios` to follow naming conventions of `<function>-<subteam>`
- finalize first integration with `volume-ios` (peep [`conner/analytics`](https://github.com/cuappdev/volume-ios/tree/conner/analytics) branch)
- integrate with pear, then other apps (iterating if necessary)

# Open questions
- should we `.gitignore` `Pods` to prevent massive diffs on cocoapod changes?
- should we try to see what Firebase packages are actually used to slim down our `Podfile.lock`? (I just went off of the official Firebase tutorial which seems to import **everything**)

### CODE REVIEW

**Comment by danielVebman** [comment]
I would put Pods in .gitignore in part because firebase packages are huge. Also, like you hinted at, it makes reviewing PRs basically impossible, especially for a highly maintained pod that is likely to change relatively often. I know that I'm not sifting through 595 files. I'm bound to miss the significant changes anyway.

**Review by camfruss — APPROVED** [review]

File: AppDevAnalytics/Classes/Analytics.swift [hunk 1]
```diff
@@ -6,18 +6,30 @@
...
+    var parameters: [String: Any]? { get }
+}
+
+public extension Event {
+    var parameters: [String: Any]? {
+        nil
+    }
+}
+
+public class AppDevAnalytics {
```
Comment (danielVebman) [inline AppDevAnalytics/Classes/Analytics.swift]: If you're not adding blank newlines after ever object/extension declaration and before its closing `}`, then Delete the newline.

File: AppDevAnalytics/Classes/Analytics.swift [hunk 2]
```diff
@@ -6,18 +6,30 @@
...
 
-    static let shared = AppDevAnalytics()
+    static public let shared = AppDevAnalytics()
 
     private init() {}
 
-    func log(_ payload: Payload) {
+    static public func logEvent(_ event: Event) {
```
Comment (danielVebman) [inline AppDevAnalytics/Classes/Analytics.swift]: `static public func log(event: Event) {` is more idiomatic and concise. If it is clear from context, `log(_ event: Event)` would be even better.

File: Podfile [hunk 3]
```diff
@@ -1,12 +1,11 @@
 # Uncomment the next line to define a global platform for your project
-platform :ios, '9.0'
+# platform :ios, '9.0'
```
Comment (danielVebman) [inline Podfile]: I think such an old Swift version is overkill and results in older framework versions.

**Review by danielVebman — APPROVED** [review]
Nice 🤩 just a couple small comments.

**Pull request merged**